design-documents/catalog-domain/simple-products.md

```diff
@@ -0,0 +1,13 @@
+# Simple product
+* Simple product is a base unit of sale.
+* Simple product has SKU attribute which uniquely represents a product at the storefront.
+* Simple product should have attributes which represent it at the listings and PDP.
+* Simple product may have images.
+* Simple product may have inventory.
+* Simple product may appear in listings and has PDP.
+* Simple product should have an URL. 
```

Comment:
disagree. product has to know how to expose reference on itself with no extra calls.  as for me, this is natural that product has URL.

Reply:
Product is part of domain model and does not need to know how (and if) it will be displayed.
For example, if Product service will be used for data only and synced to Amazon or other 3rd-party system, why does it need a URL? URL is a characteristic of PDP, which is a part of presentation layer, and uses data from Product model.
I'm not saying about extra calls. Just that product (in my opinion) doesn't "have URL", but that "there is a URL associated with the product". All this data (URL, price, inventory information) can be provided by the Catalog service if needed though.

Reply:
The main purpose of catalog service is to be a data source for PDP and PLP. 
PLP requires to know about URL to make a redirect on PDP

Reply:
I _think_ I agree with Olga's idea here (expanding on my other comments about URL).

> product has to know how to expose reference on itself with no extra calls

This is definitely a need for the client-side, but other platforms solve this in a different way. Options exist, including signals in URLs that include an entity type and identifier. Example:

`mystore.dev/product/foo`
`mystore.com/{entity-type}/{identifier}`

Not saying we should choose one right now, but I don't think we have a strict technical requirement for the product model to know about URLs.

Reply:
Example scenario: Imagine a multi store setup with a single Magento instance - 1 product catalog, 2 UIs. The store started out with one headless UI created by a 3rd party, and then later decides to spin up the second store view using a different 3rd party's headless solution.

Those 2 UIs could have different URL structures, which would require that (in this model) our data layer now keep a _list_ of URLs per product. I'm not convinced the data layer needs to know about these cases